packages/state-transition/src/util/slotInterval.ts

```diff
@@ -0,0 +1,39 @@
+import {INTERVALS_PER_SLOT} from "@lodestar/params";
+import {Slot} from "@lodestar/types";
+/**
+ * SlotInterval defines intervals within a slot.
+ * For example, attestation propagation happens at the beginning of interval 1 (second interval) of the slot
+ * whereas block propagation happens at the beginning of interval 0 (first interval) of the slot.
+ *
+ * This can also be interpret as seconds into the slot.
+ * For example, block propagation happens at (0 * SECONDS_PER_SLOT / INTERVALS_PER_SLOT) seconds into the slot
+ * and attestation propagation happens (1 * SECONDS_PER_SLOT / INTERVALS_PER_SLOT) seconds into the slot
+ *
+ * Some intervals might have several validator actions eg. aggregate and sync aggregate both happening at the beginning
+ * inteval 2.
+ */
+/* eslint-disable @typescript-eslint/no-duplicate-enum-values */
+export enum SlotInterval {
+  BLOCK_PROPAGATION = 0,
```

Comment:
Wondering if the name here is accurate because the interval includes block production (local and builder) while propagation (via gossipsub) of the block a lot of the time happens at 2 seconds or later into the slot.